sdk/assemblyscript/src/assembly/neo4j.ts

```diff
@@ -154,30 +155,28 @@ export class Record {
     return map;
   }
 
-  __INITIALIZE(): this {
-    return this;
-  }
 
-  __SERIALIZE(): string {
-    let result = "{";
-    for (let i = 0; i < this.keys.length; i++) {
-      const keyJson = JSON.stringify(this.keys[i]);
-      result += `${keyJson}:${this.values[i]}`;
-      if (i < this.keys.length - 1) {
-        result += ",";
-      }
+  @serializer
+  serialize(self: Record): string {
+    let out = "{";
+    const end = self.keys.length - 1;
+    for (let i = 0; i < end; i++) {
+      const key = JSON.stringify(self.keys[i]);
+      out += key + ":" + self.values[i] + ",";
     }
-    return result + "}";
+
+    if (end >= 0) {
+      const key = JSON.stringify(self.keys[end]);
+      out += key + ":" + self.values[end];
+    }
+
+    return out + "}";
```

Comment:
This seems like a micro-optimization.  It's easier to read and should perform roughly the same with the previous implementation, IMHO.